src/relay/azext_relay/commands.py

```diff
@@ -0,0 +1,78 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands import CliCommandType
+from azext_relay._client_factory import (namespaces_mgmt_client_factory, wcfrelay_mgmt_client_factory, hydrid_connections_mgmt_client_factory)
+from azext_relay.custom import empty_on_404
+
+
+def load_command_table(self, _):
+
+    relay_namespace_util = CliCommandType(
+        operations_tmpl='azext_relay.relay.operations.namespaces_operations#NamespacesOperations.{}',
+        client_factory=namespaces_mgmt_client_factory,
+        client_arg_name='self'
```

Comment:
Why are you using self?  You shouldn't need this at all. In your custom commands, the client arg name is 'client'.

Reply:
if removed, gives the positional parameter missing error

Reply:
Apparently it's due to being an extension. Hopefully I can remove the need for this in a future release.